### PR TITLE
[Task] All samples unified switch to annotation + yaml configuration …

### PR DESCRIPTION
**Pull request by cnzakii**
## What is the purpose of the change
[Task] All samples unified switch to annotation + yaml configuration [#13860](https://github.com/apache/dubbo/issues/13860)

## Brief changelog
Convert all `application.properties` into `application.yml`

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — COMMENTED** [review]
@CrazyHZM PTAL

**Comment by CrazyHZM** [comment]
@cnzakii Please fix the ci.

**Comment by cnzakii** [comment]
@CrazyHZM 
The reasons causing the CI checks to fail do not appear to be caused by this PR:
1. The CI check for Dubbo 3.2 passed without issues.
2. The failures in `Dubbo 3.3 / Integration Test (Java17, Job4)` and `Dubbo 3.3 / Integration Test (Java21, Job4)` were due to timeouts in the `dubbo-samples-callback` module, but this PR did not modify that module.

4. The failure in `Dubbo 3.3 / Integration Test (Java21, Job3)` was due to `log4j-slf4j2-impl cannot be present with log4j-to-slf4j`, which is not caused by the configuration files.

File: 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-protocol-consumer/src/main/resources/application.yml [hunk 1]
```diff
@@ -13,15 +13,17 @@
...
-
-# Enable token verification for each invocation
-dubbo.provider.token=true
-
-# Specify the registry address
-dubbo.registry.address=nacos://localhost:8848?username=nacos&password=nacos
-# dubbo.registry.address=nacos://${nacos.address:localhost}:8848?username=nacos&password=nacos
-
-# Specify the port of eDubbo protocol
-dubbo.provider.protocol=edubbo
```
Comment (CrazyHZM) [inline 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-protocol-consumer/src/main/resources/application.yml:27]: Please let me know why this config is missing. This configuration is a sample of core content. Please check all the changes.

File: 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-filter-provider/src/main/resources/application.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        port: 20881
+    provider:
+        # Apply AppendedFilter
+        filter: appended
+        # Enable token verification for each invocation
+        token: true
+    registry:
+        # Specify the registry address
+        address: nacos://localhost:8848?username=nacos&password=nacos
+#        address: nacos://localhost:8848?username=nacos&password=nacos
```
Comment (CrazyHZM) [inline 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-filter-provider/src/main/resources/application.yml:31]: Is this configuration file only a new, no-removed properties file?

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

**Comment by CrazyHZM** [comment]
@cnzakii For new contributors to the community, we would prefer to see as small a PR as possible for one or a few modules, as this is easier to review and can be merged faster.

File: 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-filter-provider/src/main/resources/application.yml (hunk 2 above)
Comment (cnzakii) [reply 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-filter-provider/src/main/resources/application.yml:31]: No, this is also replacing `application.properties` with `application.yml`
old:
image
image
new:
image

File: 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-protocol-consumer/src/main/resources/application.yml (hunk 1 above)
Comment (cnzakii) [reply 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-protocol-consumer/src/main/resources/application.yml:27]: Sorry, this is a mistake in my work, I will fix it and check if there are any similar errors.

Comment (cnzakii) [reply 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-protocol-consumer/src/main/resources/application.yml:27]: My working methods are all the same：
1. Copy the `application.properties` copy
2. Use the `Convert YAML and Properties File` plugin to convert the `application.properties` into `application.yml`
3. Move the annotations in the copy to `application.yml`
5. delete the copy.

**Comment by cnzakii** [comment]
> @cnzakii For new contributors to the community, we would prefer to see as small a PR as possible for one or a few modules, as this is easier to review and can be merged faster.

OK, I get it, thanks for reminding me

File: 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-filter-provider/src/main/resources/application.yml (hunk 2 above)
Comment (CrazyHZM) [reply 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-filter-provider/src/main/resources/application.yml:31]: @cnzakii My meaning is we should remove the application.properties.

Comment (cnzakii) [reply 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-filter-provider/src/main/resources/application.yml:31]: @CrazyHZM 
My reply above is explaining that I have deleted `application.properties`.
But I don’t know why it is not shown here that `application.properties` has been deleted

Comment (CrazyHZM) [reply 10-task/dubbo-samples-extensibility/dubbo-samples-extensibility-filter-provider/src/main/resources/application.yml:31]: Ok, I checked your branch, and it is ok; it seems to be a GitHub issue.

**Comment by chickenlj** [comment]
Thanks @cnzakii , we really appreciate your contribution. I agree with @CrazyHZM lease one module each time.

**Comment by cnzakii** [comment]
OK, I will close this "massive" pull request and submit new pull requests module by module.

**Pull request closed without merging**